### ISSUE TITLE
Add support for GroupByAppend

### ISSUE THREAD

**Issue by ashishkf** [doug-martin/goqu#275]
**Is your feature request related to a problem? Please describe.**
Unlike Select, Order, GroupBy have only GroupBy and SetGroupBy but not GroupByAppend.
Without this, we cannot add additional group by clauses - we need to get existing ones, do the append outside and then call SetGroupBy.

**Describe the solution you'd like**
A method on selectClauses GroupByAppend as follows:
```
func (c *selectClauses) GroupByAppend(cl ColumnListExpression) SelectClauses {
     ret := c.clone()
    ret.groupBy = ret.groupBy.Append(cl.Columns()...)
   return ret
}
```
Also, corresponding method on SelectDataset as follows:
```
func (sd *SelectDataset) GroupByAppend(groupBy ...interface{}) *SelectDataset {
   return sd.copy(sd.clauses.GroupByAppend(exp.NewColumnListExpression(groupBy...)))
}
```
**Dialect**
- [x] postgres
- [x] mysql
- [x] sqlite3

**Comment by ashishkf** [comment]
https://github.com/doug-martin/goqu/pull/276

**Comment by doug-martin** [comment]
Released in `v9.14.0` thank you for your PR!
